base/compiler/ssair/inlining.jl

```diff
@@ -1016,20 +1016,23 @@ is_builtin(s::Signature) =
     isa(s.f, Builtin) ||
     s.ft ⊑ Builtin
 
-function inline_invoke!(ir::IRCode, idx::Int, sig::Signature, invoke_data::InvokeData, state::InliningState, todo::Vector{Pair{Int, Any}})
+function inline_invoke!(ir::IRCode, idx::Int, sig::Signature, info::InvokeCallInfo,
+        state::InliningState, todo::Vector{Pair{Int, Any}})
     stmt = ir.stmts[idx][:inst]
     calltype = ir.stmts[idx][:type]
-    method = invoke_data.entry
-    (metharg, methsp) = ccall(:jl_type_intersection_with_env, Any, (Any, Any),
-            sig.atype, method.sig)::SimpleVector
-    methsp = methsp::SimpleVector
-    match = MethodMatch(metharg, methsp, method, true)
-    et = state.et
-    result = analyze_method!(match, sig.atypes, et, state.caches, state.params, calltype)
-    handle_single_case!(ir, stmt, idx, result, true, todo)
-    if et !== nothing
-        intersect!(et, WorldRange(invoke_data.min_valid, invoke_data.max_valid))
+
+    if !info.match.fully_covers
+        # XXX: We could union split this
```

Comment:
Doesn't seem like a bug (e.g. `XXX`)? Also not true after this PR?

Reply:
Well, we don't have a standard notion of `XXX`. In this case it's just here to let people know that'd it'd be ok to do, but I didn't implement it. If can use `TODO` if you'd prefer. The union splitting that could be accomplished here is inlining the signature check, which isn't really a union split per-se, but the union split code kinda does as a side effect.

Reply:
True, XXX just looks like flag of bad code, where this is fine